stdlib/src/utils/string_slice.mojo

```diff
@@ -1423,7 +1422,7 @@ fn _split_impl[
     if sep_len == 0:
         var ptr = src_str.unsafe_ptr()
         var iterator = _StringSliceIter[
-            _lit_mut_cast[__origin_of(src_str), False].result
+            _lit_mut_cast[False, __origin_of(src_str)].result
```

Comment:
You could use `ImmutableOrigin.cast_from[__origin_of(src_str)].result`.

Reply:
Turns out I could've just used the `O: ImmutableOrigin` param :v